include/ringmesh/mesh/cartesian_grid.h

```diff
@@ -168,17 +232,83 @@ namespace RINGMesh
             return coords;
         }
 
+        index_t nb_cells() const
+        {
+            return nb_total_cells_;
+        }
+
+        GEO::AttributesManager& attributes_manager() const
+        {
+            return attributes_manager_;
+        }
+
+        double cell_volume() const
+        {
+            double cell_volume{ 1. };
+            double vector_norm;
+            for( auto i : range( DIMENSION ) )
+            {
+                vector_norm = 0;
+                for( auto j : range( DIMENSION ) )
+                {
+                    vector_norm +=
+                        cartesian_frame_[i][j] * cartesian_frame_[i][j];
+                }
+                cell_volume *= std::sqrt( vector_norm );
+            }
+            return cell_volume;
```

Comment:
For this function, I wound use the vecn<DIMENSION> length() (or length2() function. It is clearer and shorter